bin/generators/plugins/install.js

```diff
@@ -0,0 +1,209 @@
+const eg = require('../../eg');
+const PluginInstaller = require('../../../lib/plugin-installer');
+
+module.exports = class extends eg.Generator {
+  constructor (args, opts) {
+    super(args, opts);
+
+    this.installer = PluginInstaller.create();
+
+    this.pluginOptions = null;
+
+    this.enablePlugin = false;
+    this.addPoliciesToWhitelist = false;
+
+    this.configureCommand({
+      command: 'install <package> [options]',
+      description: 'Install a plugin',
+      builder: yargs =>
+        yargs
+          .usage(`Usage: $0 ${process.argv[2]} install <package> [options]`)
+          .example(`$0 ${process.argv[2]} install express-gateway-plugin-url-rewrite`)
+          .option('n', {
+            alias: 'enable',
+            describe: 'Enable plugin in system config',
+            demandOption: false,
+            type: 'boolean'
+          })
+          .option('g', {
+            alias: 'update-gateway-config',
+            describe: 'Update policy whitelist in gateway config',
+            demandOption: false,
+            type: 'boolean'
+          })
+          .option('o', {
+            alias: 'option',
+            describe: 'User property in the form [-p \'foo=bar\']',
+            demandOption: false,
+            type: 'string'
+          })
+    });
+  }
+
+  initializing () {
+    this.enablePlugin = this.argv.n || null;
+    this.addPoliciesToWhitelist = this.argv.g || null;
+
+    let optionValues = [];
+
+    const argv = this.argv;
+    if (argv.o) {
+      optionValues = Array.isArray(argv.o) ? argv.o : [argv.o];
+    }
+
+    let hasInvalidProperty = false;
+
+    if (optionValues.length) {
+      this.pluginOptions = {};
+
+      const self = this;
+      optionValues.forEach(o => {
+        const equalIndex = o.indexOf('=');
+
+        if (equalIndex === -1 || equalIndex === o.length - 1) {
+          this.log.error('invalid option:', o);
+          hasInvalidProperty = true;
+          return;
+        }
+
+        const key = o.substring(0, equalIndex);
+        const value = o.substring(equalIndex + 1);
+
+        self.pluginOptions[key] = value;
+      });
+    }
+
+    if (hasInvalidProperty) {
+      return;
+    }
+
+    return this.installer.runNPMInstallation({
```

Comment:
../express-gateway/bin/index.js plugin install ExpressGateway/express-gateway-plugin-example
does not save in the package.json for npm 4.6 most likely it will save for npm5.4 

can you add --save option here ?

Reply:
Done-zo.